src/emit/mod.rs

```diff
@@ -263,13 +283,13 @@ pub trait TargetRuntime<'a> {
         encoded_args: BasicValueEnum<'b>,
         encoded_args_len: BasicValueEnum<'b>,
         gas: IntValue<'b>,
-        value: Option<IntValue<'b>>,
-        salt: Option<IntValue<'b>>,
-        seeds: Option<(PointerValue<'b>, IntValue<'b>)>,
+        contract_args: ContractArgs<'b>,
         ns: &Namespace,
+        loc: Loc,
     );
 
     /// call external function
+    #[allow(clippy::too_many_arguments)]
     fn external_call<'b>(
```

Comment:
I think we can use `struct ContractArgs` here too. What do you think @seanyoung?

Reply:
Ideally yes, however we're trying to get the PR over the line so it can be merged, so let's leave this up to @salaheldinsoliman